pkg/build/admission/buildpodutil.go

```diff
@@ -56,10 +56,50 @@ func SetBuild(a admission.Attributes, build *buildapi.Build, groupVersion unvers
 	if err != nil {
 		return err
 	}
+
 	err = setBuildInPod(build, pod, groupVersion)
 	if err != nil {
 		return admission.NewForbidden(a, fmt.Errorf("unable to set build in pod: %v", err))
 	}
+
+	err = applyBuildLogLevel(build, pod)
+	if err != nil {
+		return admission.NewForbidden(a, fmt.Errorf("unable to set loglevel in pod: %v", err))
+	}
+
+	return nil
+}
+
+// applyBuildLogLevel extracts BUILD_LOGLEVEL from the Build environment
+// and feeds it as an argument to the Pod's entrypoint. The BUILD_LOGLEVEL
+// environment variable may have been set in multiple ways: a default value,
+// by a BuildConfig, or by the BuildDefaults admission plugin. In this method
+// we finally act on the value by injecting it into the Pod.
+func applyBuildLogLevel(build *buildapi.Build, pod *kapi.Pod) error {
+
+	var envs []kapi.EnvVar
+
+	// Check whether the build strategy supports --loglevel parameter.
+	switch {
+	case build.Spec.Strategy.DockerStrategy != nil:
+		envs = build.Spec.Strategy.DockerStrategy.Env
+	case build.Spec.Strategy.SourceStrategy != nil:
+		envs = build.Spec.Strategy.SourceStrategy.Env
+	default:
+		// The build strategy does not support --loglevel
+		return nil
+	}
+
+	buildLogLevel := "1" // The ultimate default for the builder's loglevel if no actor sets BUILD_LOGLEVEL in the Pod
```

Comment:
1 was just my guess. Parsing "most users do not need log level 2 output from the build" left some room for interpretation. I will change to 0.